tests/unit/test_plugin_calculator.py

```diff
@@ -0,0 +1,87 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# pylint: disable=missing-module-docstring
+
+from mock import Mock
+from parameterized.parameterized import parameterized
+from searx import plugins
+from tests import SearxTestCase
+from .test_utils import random_string
+
+from .test_plugins import get_search_mock
+
+
+class PluginCalculator(SearxTestCase):  # pylint: disable=missing-class-docstring
+    def setUp(self):
+        self.store = plugins.PluginStore()
+        plugin = plugins.load_and_initialize_plugin('searx.plugins.calculator', False, (None, {}))
+        self.store.register(plugin)
+
+    def test_plugin_store_init(self):
+        self.assertEqual(1, len(self.store.plugins))
+
+    def test_single_page_number_true(self):
+        request = Mock(remote_addr='127.0.0.1')
+        search = get_search_mock(query=random_string(10), pageno=2)
+        result = self.store.call(self.store.plugins, 'post_search', request, search)
+        self.assertTrue(result)
+        self.assertNotIn('calculate', search.result_container.answers)
+
+    def test_long_query_true(self):
+        request = Mock(remote_addr='127.0.0.1')
+        search = get_search_mock(query=random_string(101), pageno=1)
+        result = self.store.call(self.store.plugins, 'post_search', request, search)
+        self.assertTrue(result)
+        self.assertNotIn('calculate', search.result_container.answers)
+
+    def test_alpha_true(self):
+        request = Mock(remote_addr='127.0.0.1')
+        search = get_search_mock(query=random_string(10), pageno=1)
+        result = self.store.call(self.store.plugins, 'post_search', request, search)
+        self.assertTrue(result)
+        self.assertNotIn('calculate', search.result_container.answers)
+
+    @parameterized.expand(
+        [
+            ("1+1", "2"),
+            ("1-1", "0"),
+            ("1*1", "1"),
+            ("1/1", "1"),
+            ("1**1", "1"),
+            ("1^1", "1"),
+            # en-US
```

Comment:
nit: although it's more verbose, we could just remove the default parameter assignment and have en-US for each rest. Then we have no need for comments. But this is non blocking and we can merge anyway.

Reply:
Good point :+1 .. I modified the test sources [diff](https://github.com/searxng/searxng/compare/f3541cd4dc8e15783cf9685fbdbdac1d8b0a1bf1..2c20f41e6077335bc9f0dda70c2bada16fe15fb5)